### PR TITLE
Adapt Onboarding to Figma prototypes

### LINKED ISSUE #273: Implement onboarding screens like in the Figma designs

### PR DESCRIPTION
**Pull request by SebastianWagner2**
Closes #273 

- [x] I have read the [code-style guidelines](https://github.com/hpi-dhc/PharMe/blob/main/docs/FLUTTER_STYLE.md) and confirmed that this PR upholds said guidelines

---

### CODE REVIEW

File: app/lib/onboarding/pages/page.dart [hunk 1]
```diff
@@ -0,0 +1,267 @@
+import 'package:url_launcher/url_launcher.dart';
+
+import '../../../common/models/metadata.dart';
+import '../../../common/module.dart' hide MetaData;
+
```
Comment (kolioOtSofia) [inline app/lib/onboarding/pages/page.dart:5]: I would change the file name to onboarding.dart, page.dart is also annoying when you are searching for the page with ctr + p

File: app/lib/onboarding/pages/page.dart [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+          Positioned(
+            bottom: 16,
+            right: 16,
+            child: _buildNextButton(
+              context,
+              pageController,
+              currentPage.value == _pages.length - 1,
+            ),
+          ),
+        ]),
```
Comment (kolioOtSofia) [inline app/lib/onboarding/pages/page.dart:113]:
```suggestion
        child: Stack(
        alignment: Alignment.topCenter,
        children: [
          Positioned.fill(
            child: PageView(
              controller: pageController,
              onPageChanged: (newPage) => currentPage.value = newPage,
              children: _pages,
            ),
          ),
          Positioned(
            bottom: 80,
            child: Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: _buildPageIndicator(context, currentPage.value),
            ),
          ),
          Positioned(
            bottom: 16,
            right: 16,
            child: _buildNextButton(
              context,
              pageController,
              currentPage.value == _pages.length - 1,
            ),
          ),
        ],
      ),
```
Trailing comma in the end

File: app/lib/onboarding/pages/page.dart [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    bool isLastPage,
+  ) {
+    return TextButton(
+      onPressed: () {
+        if (isLastPage) {
+          if (_isLoggedIn) {
+            context.router.pop();
+          } else {
+            context.router.replace(LoginRouter());
+          }
```
Comment (kolioOtSofia) [inline app/lib/onboarding/pages/page.dart:151]:
```suggestion
          _isLoggedIn ? context.router.pop() : context.router.replace(LoginRouter());
  
```
I would use tenary operator for the inner statement, it's smh weird with this if-else just for one statement to be executed in the respective case

**Review by kolioOtSofia — CHANGES_REQUESTED** [review]

File: app/lib/onboarding/pages/page.dart (hunk 1 above)
Comment (SebastianWagner2) [reply app/lib/onboarding/pages/page.dart:5]: just for the consistency im reverting this change :P

**Review by kolioOtSofia — APPROVED** [review]
LGTM

**Pull request merged**